opm/simulators/flow/Transmissibility_impl.hpp

```diff
@@ -243,296 +244,235 @@ update(bool global, const TransUpdateQuantities update_quantities,
+        {
+            return {
+                halfComputer(faceAreaNormal,
+                             inside.faceIdx,
+                             distanceVector_(inside.faceCenter, inside.elemIdx),
+                             prop1),
+                halfComputer(faceAreaNormal,
+                             outside.faceIdx,
+                             distanceVector_(outside.faceCenter, outside.elemIdx),
+                             prop2)
+            };
+        };
+
+        auto computeHalfMean = [&inside, &outside, &computeHalf, &ntg, &harmonicMean]
+                               (const auto& halfComputer, const auto& prop)
+        {
+            auto half = computeHalf(halfComputer, prop[inside.elemIdx], prop[outside.elemIdx]);
+            applyNtg_(half[0], inside, ntg);
+            applyNtg_(half[1], outside, ntg);
+
+            //TODO Add support for multipliers
+            return harmonicMean(half[0], half[1]);
+        };
 
+        unsigned boundaryIsIdx = 0;
+        for (const auto& intersection : intersections(gridView_, elem)) {
             // deal with grid boundaries
             if (intersection.boundary()) {
                 // compute the transmissibilty for the boundary intersection
                 const auto& geometry = intersection.geometry();
-                const auto& faceCenterInside = geometry.center();
+                inside.faceCenter = geometry.center();
 
-                auto faceAreaNormal = intersection.centerUnitOuterNormal();
+                faceAreaNormal = intersection.centerUnitOuterNormal();
                 faceAreaNormal *= geometry.volume();
 
-                Scalar transBoundaryIs;
-                computeHalfTrans_(transBoundaryIs,
-                                  faceAreaNormal,
-                                  intersection.indexInInside(),
-                                  distanceVector_(faceCenterInside,
-                                                  elemIdx),
-                                  permeability_[elemIdx]);
+                Scalar transBoundaryIs =
+                    computeHalfTrans_(faceAreaNormal,
+                                      intersection.indexInInside(),
+                                      distanceVector_(inside.faceCenter, inside.elemIdx),
+                                      permeability_[inside.elemIdx]);
 
                 // normally there would be two half-transmissibilities that would be
                 // averaged. on the grid boundary there only is the half
                 // transmissibility of the interior element.
-                unsigned insideCartElemIdx = cartMapper_.cartesianIndex(elemIdx);
-                applyMultipliers_(transBoundaryIs, intersection.indexInInside(), insideCartElemIdx, transMult);
-                transBoundary_[std::make_pair(elemIdx, boundaryIsIdx)] = transBoundaryIs;
+                inside.cartElemIdx = cartMapper_.cartesianIndex(inside.elemIdx);
```

Comment:
@aritorto can you explain the reason for this lookup not going through lookupdata?

Reply:
If I remember correctly, we do not support multipliers - yet - for CpGrid with lgrs. I'll have a deeper look though. That would be my first answer!

Reply:
As far as I can see, the difference here is that the intersection is on the boundary. We will continue the intersection loop inside this if, so we will never hit the multiplier code in this code path.

Reply:
Just double-checked that, for a CpGrid with Lgrs, cartMapper_.cartesianIndex(inside.elemidx) returns the Cartesian index of the parent cell. So, it's correct in the end. 

When we only need the Cartesian Index of an element on the leaf grid, then we can just use its Cartesian mapper. If we want to know the value of a field property of a leaf element, then it's more convenient to use the lookup via lookupCartesianData. 

I hope this helps!

Reply:
What I am trying to understand is whether or not I can move the lookup for the inside cell outside the intersection loop so that the lookup is done only once and not once per intersection. If it is safe to replace this call by a call through lookupdata then that is the case, and I can move https://github.com/OPM/opm-simulators/blob/master/opm/simulators/flow/Transmissibility_impl.hpp#L310 outside the loop and remove this line.

Reply:
I see your point. Indeed, it's much better and safe to replace this call by a call through lookup**Cartesian**Data.

Reply:
yeah sorry for my shorting, I of course mean lookupCartesianData. thx!